src/rgw/rgw_rados.cc

```diff
@@ -9013,3 +9013,131 @@ librados::Rados* RGWRados::get_rados_handle()
   }
 }
 
+int encrypt(unsigned char *plaintext, int plaintext_len, unsigned char *key,
+  unsigned char *iv, unsigned char *ciphertext)
+{
+  EVP_CIPHER_CTX *ctx;
+
+  int len;
+
+  int ciphertext_len;
+
+  /* Create and initialise the context */
+  if(!(ctx = EVP_CIPHER_CTX_new())) handleErrors();
+
```

Comment:
do we need a new context every time ? I am not sure whether this is needed every time or  not and what is the cost